ql/instruments/forwardrateagreement.cpp

```diff
@@ -61,8 +61,72 @@ namespace QuantLib {
     : ForwardRateAgreement(valueDate, index->maturityDate(valueDate), type, strikeForwardRate,
                            notionalAmount, index, std::move(discountCurve), true) {}
 
+    ForwardRateAgreement::ForwardRateAgreement(Position::Type type,
+                                               const Date& valueDate,
+                                               Rate strikeForwardRate,
+                                               Real notionalAmount,
+                                               const ext::shared_ptr<IborIndex>& index,
+                                               Handle<YieldTermStructure> discountCurve,
+                                               bool useIndexedCoupon)
+    : fraType_(type), valueDate_(valueDate), notionalAmount_(notionalAmount),
+      index_(index), discountCurve_(std::move(discountCurve)),
+      useIndexedCoupon_(useIndexedCoupon), 
+      dayCounter_(index->dayCounter()), calendar_(index->fixingCalendar()),
+      businessDayConvention_(index->businessDayConvention()),
+      maturityDate_(index->maturityDate(valueDate)) {
+
+        QL_REQUIRE(notionalAmount > 0.0, "notionalAmount must be positive");
+
+        registerWith(Settings::instance().evaluationDate());
+        registerWith(discountCurve_);
+        registerWith(index_);
+        
+        maturityDate_ = calendar_.adjust(maturityDate_, businessDayConvention_);
+        QL_REQUIRE(valueDate_ < maturityDate_, "valueDate must be earlier than maturityDate");
+        strikeForwardRate_ = InterestRate(strikeForwardRate,
+                                          index->dayCounter(),
+                                          Simple, Once);
+    }
+
+    ForwardRateAgreement::ForwardRateAgreement(const Date& valueDate,
+                                               const Date& maturityDate,
+                                               Position::Type type,
+                                               Rate strikeForwardRate,
+                                               Rate referenceRate,
```

Comment:
Ah, i think this is the wrong approach. For the case with custom dates, we still want to get the rate from a YieldTermStructure. If i understand correctly, here you are requiring a fixed rate be supplied when the object is constructed.

In general, the way QuantLib works is that you define a graph of objects, where instruments depend on indexes, which depend on term structures, which depend on helpers, which depend on quotes. Then, you can update the quotes with some market data, and read out the prices for instruments implied by the market data. If you require a fixed rate like this, you make it impossible to construct that graph.

If you want to take Luigi's suggested approach of not taking an index, then i think you should take a `Handle<YieldTermStructure>` instead.